### PR TITLE
make x86_64-pc-solaris the default target for x86-64 Solaris

### PR DESCRIPTION
**Pull request by kulikjak**
This change makes `x86_64-pc-solaris` the default compilation target for x86-64 Solaris/Illumos (based on [this exchange](https://github.com/rust-lang/rust/issues/68214#issuecomment-748042054) with @varkor).

I tried several ways of doing this (leveraging the alias support added with #61761 and improved/fixed with #80073) and found out that cross-compilation to the new one is by far the simplest way of doing this. It can be achieved by adding the following arguments: `--build x86_64-sun-solaris --host x86_64-pc-solaris --target x86_64-pc-solaris` and enabling the cross compilation with `PKG_CONFIG_ALLOW_CROSS=1` environment variable.

I also removed alias support altogether - `x86_64-pc-solaris` and `x86_64-sun-solaris` are now two separate targets. The problem with aliases is that even if rust internally knows that two are the same, other tools building with rust don't know that, resulting in build issues like the one with firefox mentioned [here](https://github.com/rust-lang/rust/issues/68214#issuecomment-746144229). I think that once the dust settles and `x86_64-pc-solaris` becomes the default, `x86_64-sun-solaris` can be removed.

If you agree with the above, I have two subsequent questions:
1. Is there a preferred way to display deprecation warnings when `x86_64-sun-solaris` is passed into the compiler as an argument? I am not sure whether target deprecation was done before.
2. Where would be the best way to document this change for those using rust on Solaris? Without the cross-compilation arguments (used once to build a new version), the build won't work. Should I add it into [RELEASES.md](https://github.com/rust-lang/rust/blob/master/RELEASES.md)?

Thanks!

### CODE REVIEW

File: src/doc/rustc/src/platform-support.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ target | std | host | notes
 `x86_64-fuchsia` | ✓ |  | 64-bit Fuchsia
 `x86_64-linux-android` | ✓ |  | 64-bit x86 Android
 `x86_64-rumprun-netbsd` | ✓ |  | 64-bit NetBSD Rump Kernel
-`x86_64-sun-solaris` | ✓ |  | 64-bit Solaris 10/11, illumos
+`x86_64-pc-solaris` | ✓ |  | 64-bit Solaris 10/11, illumos
```
Comment (nagisa) [inline src/doc/rustc/src/platform-support.md:137]: Doesn't this require changes to CI? Right now we produce artifacts for `x86_64-sun-solaris`.

Comment (kulikjak) [reply src/doc/rustc/src/platform-support.md:137]: You are right. The previous target will still work, but the artifacts produced should be for the preferred/correct target. I will look into it.

Comment (kulikjak) [reply src/doc/rustc/src/platform-support.md:137]: Hopefully, I changed it correctly, although I am not a Docker user, so maybe I missed/misunderstood something.

**Comment by varkor** [comment]
Thanks for looking into this! I had a quick look through, but I'd rather someone more familiar reviewed this.

r? @Mark-Simulacrum

**Comment by Mark-Simulacrum** [comment]
r? @nagisa perhaps

I think it probably makes sense to try and put out some kind of call to solaris users (e.g., on users.rust-lang.org) to provide some feedback here, as I myself don't really know why sun was previously chosen (and if or why that was wrong). Using the same name LLVM uses does seem better to me, though, in general. I think for tier 2/3 targets we shouldn't guarantee anything really so renaming shouldn't be a problem.

**Comment by nagisa** [comment]
You will need to rebase (rather than merge) before this can be approved.

The changes here overall seem okay to me (assuming @Mark-Simulacrum looked at the CI-related changes), though it does definitely make sense to me to broadcast this change on users.rlo once this is approved.

**Comment by Mark-Simulacrum** [comment]
Yes, CI changes seem pretty reasonable to me.

**Comment by nagisa** [comment]
@bors r=nagisa,Mark-Simulacrum rollup=never

cc @jclulow who recently contributed around the illumos support.

**Comment by kulikjak** [comment]
Thanks @nagisa and @Mark-Simulacrum!

**Comment by jhpratt** [comment]
This appears to be the source of my CI issues. I understand that the `x86_64-sun-solaris` target is deprecated, but it appears to have been outright _removed_ for Rust 1.52. Running `rustup target add x86_64-sun-solaris` throws an error on stable as of today.

Is this a mistake? Breaking CI without any warning seems…bad.

**Pull request merged**